records/improving_search.md

```diff
@@ -0,0 +1,135 @@
+
+Looks like a cool (modern) alternative to elasticsearch, if we wanted to go down the route of running dedicated server-side infrastructure for search.
+
+### [Stork]
+
+* Client-side with cli index builder
+* Language: Rust
+* Last release: [a year ago 1.6.0](https://crates.io/crates/stork-lib)
+* Maintainer's photo: Outside in the sun holding a hoop
+
+Looks like a cool approach to offline index building and browser-side querying. The creator announced in 2023 he is [winding down work on Stork] though is open to others adopting the code. Probably more useful to learn lessons from than try to use directly.
+
+### [Lunr]
+
+* Client-side
+* Language: Javascript
+* Last release: [three years ago 2.3.9](https://www.npmjs.com/package/lunr)
+* Maintainer's photo: Upscaled jpg black and white with a collar 
+
+"A bit like Solr, but much smaller and not as bright." is a pretty funny intro. It's not the bloat of an Apache project implemented in Java - but of the options it's one of the larger and more fully featured!
+
+### [Fuse]
+
+* Client-side
+* Language: Javascript
+* Last release: [four months ago 7.0.0](https://www.npmjs.com/package/fuse.js)
+* Maintainer's photo: Evokes a video game detective avatar
+
+### [orama]
+
+* Client-side
+* Language: Javascript
+* Last release: [today 2.0.6](https://www.npmjs.com/package/@orama/orama)
+* Maintainer's photo: Threatens to push to over and steal your skis
+
+### [flexsearch]
+
+* Client-side
+* Language: Javascript
+* Last release: [a month ago 0.7.43](https://www.npmjs.com/package/flexsearch)
+* Maintainer's photo: Upscaled crop from a holiday photo with someone else's shoulder
+
+Quite a large external interface, but does seem to include sensible options at index creation that aren't obviously available in other options. Good readme with lots of detail. The benchmarks link is dead. Eyeball code quality level looks a little lower, ancient idiosyncrasies like `prototype` usage. Actively maintained with and the [flexsearch 0.8 discussion] is encouraging.
+
+### [tinysearch]
+
+* Client-side
+* Language: Rust
+* Last release: [seven months ago 0.8.2](https://crates.io/crates/tinysearch)
+* Maintainer's photo: Professionally shot to look fun and interesting
+
+Run Rust in the browser using WASM! There's a neat [tinysearch blog post] about implementing this. I've met the author at FOSDEM and think he's solid.
+
+### [minisearch]
+
+* Client-side
+* Language: Javascript
+* Last release: [three months ago 6.3.0](https://www.npmjs.com/package/minisearch)
+* Maintainer's photo: Hanging in the park but only into the portrait
+
+Skips bundling a few options like fancy tokenising and processing but provides an interface for you to bring your own. Eyeball code quality looks pretty good, sane looking typescript and lots of inline documentation.
+
+
+## Recommendation
+
+We should try using both flexsearch with various options, and minisearch with some specific transforms we implement.
```

Comment:
But why?

Reply:
Have added a bit above that basically highlights it was easy to find how to do some stuff I want in the docs of the two recommendations here.